consensus/types/src/blob.rs

```diff
@@ -0,0 +1,223 @@
+use crate::EthSpec;
+use arbitrary::{Arbitrary, Unstructured};
+use kzg::{BlobTrait, KzgPreset};
+use rand::Rng;
+use safe_arith::SafeArith;
+use serde::{
+    de::{self, Visitor},
+    Deserialize, Deserializer, Serialize, Serializer,
+};
+use ssz::{Decode, DecodeError, Encode};
+use std::hash::{Hash, Hasher};
+use std::sync::Arc;
+use tree_hash::{PackedEncoding, TreeHash};
+
+#[derive(Clone, PartialEq, Eq)]
+pub struct SigpBlob<E: EthSpec>(Arc<<E::Kzg as KzgPreset>::Blob>);
+
+impl<E: EthSpec> SigpBlob<E> {
+    pub fn random_valid<R: Rng>(rng: &mut R) -> Result<Self, String> {
+        let mut blob_bytes = vec![0u8; <E::Kzg as KzgPreset>::BYTES_PER_BLOB];
+        rng.fill_bytes(&mut blob_bytes);
+
+        // Ensure that the blob is canonical by ensuring that
+        // each field element contained in the blob is < BLS_MODULUS
+        for i in 0..<E::Kzg as KzgPreset>::FIELD_ELEMENTS_PER_BLOB {
+            let Some(byte) = blob_bytes.get_mut(i.checked_mul(<E::Kzg as KzgPreset>::BYTES_PER_FIELD_ELEMENT).ok_or("overflow".to_string())?)  else {
+                return Err(format!("blob byte index out of bounds: {:?}", i));
+            };
+            *byte = 0;
+        }
+        let kzg_blob = <E::Kzg as KzgPreset>::Blob::from_bytes(&blob_bytes)
+            .map_err(|e| format!("failed to create blob: {:?}", e))?;
+        Ok(Self(Arc::new(kzg_blob)))
+    }
+
+    pub fn c_kzg_blob(&self) -> &<E::Kzg as KzgPreset>::Blob {
```

Comment:
How about replacing this method with 
```
impl<E: EthSpec> AsRef<<E::Kzg as KzgPreset>::Blob> for SigpBlob<E> {
    fn as_ref(&self) -> &<E::Kzg as KzgPreset>::Blob {
        self.0.as_ref()
    }
}
```

that way we don't have to have the `c_kzg` name floating around in case we swap out the dep

Reply:
As mentioned in the call, I've renamed the outer structure to `WrappedBlob` so it's clear when we're using the wrapped type. I've opted to keep `c_kzg_blob` as a function so it is clear we're getting the underlying `c_kzg::Blob`. Should we ever migrate away from `c_kzg`, it will be a simple right-click rename to update this function.

All of this is done in the name of clarity so we know where we're using the wrapped type.